azurerm/internal/services/securitycenter/security_center_workspace_resource.go

```diff
@@ -49,20 +49,19 @@ func resourceSecurityCenterWorkspace() *schema.Resource {
 			},
 
 			"workspace_id": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: azure.ValidateResourceID,
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateFunc:     validate.LogAnalyticsWorkspaceID,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
whats the reason for making the case insensitive?

Reply:
As said in PR description, Azure API will always return lower case workspace id, regardless of creation input. So without suppression it will always find differences. Current tests in master are failing, so this must have been an issue for a while.

If you have a better solution I am open to it.